packages/google_maps_flutter/google_maps_flutter_android/android/src/test/java/io/flutter/plugins/googlemaps/ClusterManagersControllerTest.java

```diff
@@ -128,6 +136,56 @@ public void AddClusterManagersAndMarkers() {
     assertEquals("Cluster should contain exactly 2 markers", 2, cluster.getSize());
   }
 
+  @Test
+  public void SelectClusterRenderer() {
+    final String clusterManagerId1 = "cm_1";
+    final String clusterManagerId2 = "cm_2";
+    final String markerId1 = "mid_1";
+    final String markerId2 = "mid_2";
+
+    when(googleMap.getCameraPosition())
+        .thenReturn(CameraPosition.builder().target(new LatLng(0, 0)).build());
+
+    ClusterManagersController controller1 =
+        spy(new ClusterManagersController(flutterApi, context, PlatformMarkerType.MARKER));
+    controller1.init(googleMap, markerManager);
+    ClusterManagersController controller2 =
+        spy(new ClusterManagersController(flutterApi, context, PlatformMarkerType.ADVANCED_MARKER));
+    controller2.init(googleMap, markerManager);
+
+    Messages.PlatformClusterManager initialClusterManager1 =
+        new Messages.PlatformClusterManager.Builder().setIdentifier(clusterManagerId1).build();
+    List<Messages.PlatformClusterManager> clusterManagersToAdd1 = new ArrayList<>();
+    clusterManagersToAdd1.add(initialClusterManager1);
+    controller1.addClusterManagers(clusterManagersToAdd1);
+
+    Messages.PlatformClusterManager initialClusterManager2 =
+        new Messages.PlatformClusterManager.Builder().setIdentifier(clusterManagerId2).build();
+    List<Messages.PlatformClusterManager> clusterManagersToAdd2 = new ArrayList<>();
+    clusterManagersToAdd2.add(initialClusterManager2);
+    controller2.addClusterManagers(clusterManagersToAdd2);
+
+    MarkerBuilder markerBuilder1 =
+        new MarkerBuilder(markerId1, clusterManagerId1, PlatformMarkerType.MARKER);
+    markerBuilder1.setPosition(new LatLng(10.0, 20.0));
+    controller1.addItem(markerBuilder1);
+
+    MarkerBuilder markerBuilder2 =
```

Comment:
Optional nit: can you modify this 1 and 2 naming convention to marker*Advanced and marker*Normal (or something else for non advanced)